fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/AggregatedMaterializedViewRewriter.java

```diff
@@ -514,11 +539,96 @@ private CallOperator getRollupAggregate(CallOperator aggCall, ColumnRefOperator
             // the rollup function is the same as origin, but use the new column as argument
             CallOperator newAggCall = (CallOperator) aggCall.clone();
             newAggCall.setChild(0, targetColumn);
+            // NOTE: Change fn's type  as 1th child has change, otherwise physical plan
+            // will still use old arg input's type.
+            newAggCall.getFunction().setArgsType(new Type[] { targetColumn.getType() });
             return newAggCall;
         }
     }
 
     private Function findArithmeticFunction(Type[] argsType, String fnName) {
         return Expr.getBuiltinFunction(fnName, argsType, Function.CompareMode.IS_IDENTICAL);
     }
+
+    // Rewrite query agg operator by rule:
+    //  - now only support rewrite avg to sum/ count
+    // TODO: maybe we can rewrite the whole query before MV's rewrite.
+    private LogicalAggregationOperator rewriteAggregationOperatorByRules(
+            ColumnRefFactory queryColumnRefFactory,
+            LogicalAggregationOperator aggregationOperator) {
+        Map<ColumnRefOperator, CallOperator> oldAggregations = aggregationOperator.getAggregations();
```

Comment:
what if the mv has a avg(x) column? then here you rewrite avg(x) into sum(x)/count(x)，then rewrite fails.
it is a badcase.

Reply:
maybe here we should consider both cases.

Reply:
- MV's `avg`  will never be used in `rollup`,  `rewriteAggregationOperatorByRules ` is only called in `rewriteForRollup`. This pr only supports avg rewrite in rollup situations.
- So for a non-rollup agg rewrite, what's your suggestion? Maybe I can fix it later.